src/traces/histogram/calc.js

```diff
@@ -217,130 +213,158 @@ module.exports = function calc(gd, trace) {
-            // and minimum bin size of them all.
-            // But manually binned traces won't be adjusted, even if the auto values
-            // are inconsistent with the manual ones (or the manual ones are inconsistent
-            // with each other).
-            minSize = getMinSize(minSize, binSpec.size);
-            minStart = Math.min(minStart, pa.r2c(binSpec.start, 0, calendar));
-            maxEnd = Math.max(maxEnd, pa.r2c(binSpec.end, 0, calendar));
-
-            // add the flag that lets us abort autobin on later traces
-            if(i) tracei._autoBinFinished = 1;
         }
 
-        // do what we can to match the auto bins to the first manual bins
-        // but only if sizes are all numeric
-        if(firstManual && isNumeric(firstManual.size) && isNumeric(minSize)) {
-            // first need to ensure the bin size is the same as or an integer fraction
-            // of the first manual bin
-            // allow the bin size to increase just under the autobin step size to match,
-            // (which is a factor of 2 or 2.5) otherwise shrink it
-            if(minSize > firstManual.size / 1.9) minSize = firstManual.size;
-            else minSize = firstManual.size / Math.ceil(firstManual.size / minSize);
-
-            // now decrease minStart if needed to make the bin centers line up
-            var adjustedFirstStart = firstManual.start + (firstManual.size - minSize) / 2;
-            minStart = adjustedFirstStart - minSize * Math.ceil((adjustedFirstStart - minStart) / minSize);
+        binOpts.size = newBinSpec.size;
+        if(!sizeFound) {
+            autoVals.size = newBinSpec.size;
+            Lib.nestedProperty(traces[0], binAttr + '.size').set(newBinSpec.size);
         }
 
-        // now go back to the autobinned traces and update their bin specs with the final values
-        for(i = 0; i < autoBinnedTraces.length; i++) {
-            tracei = autoBinnedTraces[i];
-            calendar = tracei[mainData + 'calendar'];
+        setBound('start', binOpts, newBinSpec);
+        setBound('end', binOpts, newBinSpec);
+    }
 
-            tracei._input[binAttr] = tracei[binAttr] = {
-                start: pa.c2r(minStart, 0, calendar),
-                end: pa.c2r(maxEnd, 0, calendar),
-                size: minSize
-            };
+    pos0 = trace._pos0;
+    delete trace._pos0;
 
-            // note that it's possible to get here with an explicit autobin: false
-            // if the bins were not specified.
-            // in that case this will remain in the trace, so that future updates
-            // which would change the autobinning will not do so.
-            tracei._input[autoBinAttr] = tracei[autoBinAttr];
+    // Each trace can specify its own start/end, or if omitted
+    // we ensure they're beyond the bounds of this trace's data,
+    // and we need to make sure start is aligned with the main start
+    var traceInputBins = trace._input[binAttr] || {};
```

Comment:
Why do we need to look in the input trace here?

Reply:
> Why do we need to look in the input trace here?

After the first draw we're always going to have `fullTrace.xbins.start` etc... but we can't necessarily tell whether it was explicitly specified or relinked from a previous autobin. I suppose I could have stashed that info during `supplyDefaults`, but it seemed just as effective to refer back to the input trace.

Reply:
Makes sense. Resolved.